### ISSUE TITLE
base path URL is not configurable

### ISSUE THREAD

**Issue by Mcdostone** [ElMassimo/vite_ruby#151]
**Is your feature request related to a problem? Please describe.**
I have a RoR app running on https://example.com/my-app
When I compile the assets for production, URLs used in my CSS are incorrect: expected `/my-app/vite/assets/foo.png`, got `/vite/assets/foo.png`. this issue is normal since my vite configuration is incomplete.

According to the vite documentation, the [public base path](https://vitejs.dev/guide/build.html#public-base-path) is the solution. Let's tweak the configuration:

```js
import { defineConfig } from "vite";
import RubyPlugin from "vite-plugin-ruby";

export default defineConfig({
  plugins: [RubyPlugin()],
  base: '/my-app/vite'
});
```
```bash
RAILS_ENV=production bundle exec vite build --force --debug | grep 'my-app'
# no matching lines
```
It looks like`RubyPlugin()` overwrites my custom configuration. Let's find another way.

I read through the vite-ruby doc and found the [assetHost property](https://vite-ruby.netlify.app/config/#assethost):
```json
{
  "all": {
    "sourceCodeDir": "app/assets",
    "watchAdditionalPaths": [],
    "assetHost": "/my-app"
  }
}
```
```bash
RAILS_ENV=production bundle exec vite build --force --debug | grep 'my-app'
base: '///my-app/vite/',
assetHost: '/my-app',
base: '///my-app/vite/',
BASE_URL: '///my-app/vite/',
```
We're getting close to what I would expect. However, the two leading slashes annoy me. After some research in your code the slashes are added in [config.ts:103](https://github.com/ElMassimo/vite_ruby/blob/91d2ea44873005a53e97a15e76389feccf99106f/vite-plugin-ruby/src/config.ts#L103)

**Describe the solution you'd like**
  - Be able to change the `base` property
 Or
 - remove the 2 leading slashes. Developers still have the choice to add them manually if they want to.
 ```js
 {
  "all": {
    "sourceCodeDir": "app/assets",
    "watchAdditionalPaths": [],
    "assetHost": "/my-app",                                           // or 
    "assetHost": "//my-app",      ,                                   // or
    "assetHost": "https://example.com/my-app"    
  }
}
 ```

**Comment by ElMassimo** [comment]
Hi Yann!

`assetHost` is intended for an __external__ CDN host.

Hosting a Rails app in a nested path is unusual, but I think it makes sense to add a separate `base` setting in `config/vite.json` to support this use case.

```json
{
  "all": {
     "watchAdditionalPaths": [],
  },
  "production": {
    "base": "/my-app"
  }
}
```

It would also make sense to allow configuring this setting using environment variables, like for other settings (`VITE_RUBY_BASE=/my_app`).

---

Let me know if you would be willing to rework #152 to add the `base` setting instead of modifying the `assetHost` behavior.

**Comment by ElMassimo** [comment]
Released `vite-plugin-ruby@3.0.4` and `vite_ruby@3.0.3` which add support for the [`base` setting](https://vite-ruby.netlify.app/config/#base).

Thanks!
